ion/bitstream.go

```diff
@@ -663,18 +663,7 @@ func (b *bitstream) ReadTimestamp() (Timestamp, error) {
 			return Timestamp{}, err
 		}
 
-		if nsecs > 0 {
-			fractionPrecision = 9
-
-			// Adjust fractionPrecision for each trailing zero.
-			// ie. .123456000 should have 6 fractionPrecision instead of 9
-			ns := nsecs
-			for ns > 0 && (ns%10) == 0 {
-				ns /= 10
-				fractionPrecision--
-			}
-			precision = TimestampPrecisionNanosecond
-		} else if len(fracSecsBytes) > 0 && fracSecsBytes[0] > 0xC0 && (fracSecsBytes[0]^0xC0) > 0 {
+		if len(fracSecsBytes) > 0 && fracSecsBytes[0] > 0xC0 && (fracSecsBytes[0]^0xC0) > 0 {
```

Comment:
I see what's going on - can you check against the other implementations to see if they support non-9 fractional precision?

Essentially, what this means is that if I only specify .12345 as the fraction, I've specified 5 fractional digits, therefore my fractional precision would be 5. Specifying 9 would indicate I had specified .123450000 instead.